[#182] Scan sample records for getting hbase schema

anishanainani opened the pull request:
Make sure you have checked all steps below.

### GitHub Issue
Fixes #182 


### Checklist:
<!--- Go over all the following points. Check boxes that apply to this pull request -->
- [ ] This pull request updates the documentation
- [ ] This pull request changes library dependencies
- [x] Title of the PR is of format (example) : [#25][Github] Add Pull Request Template

<!-- NOTE: lines that start with < - - ! and end with - - > are comments and will be ignored. -->
<!-- Please include the GitHub issue number in the PR title above. If an issue does not exist, please create one.-->
<!-- Example:[#25][Github] Add Pull Request Template where [#25 refers to https://github.com/paypal/gimel/issues/25] -->

### What is the purpose of this pull request?
- Get schema for hbase table automatically.
- Today the connector for HBase - open source SHC connector, requires schema to read/write. It fails if it doesnt find this property gimel.hbase.column.mapping. This feature will automatically get the schema and hence set this property.

### How was this change validated?
Spark shell

### Commit Guidelines
- [x] My commits all reference GH issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

Dee-Pac commented on gimel-dataapi/gimel-connectors/gimel-hbase-1.2/src/main/scala/com/paypal/gimel/hbase/conf/HbaseConfigs.scala:
```diff
@@ -32,8 +32,10 @@ object HbaseConfigs {
-  val hbaseUseColumnsSpecifiedFlag: String = "gimel.hbase.columns.specified.flag"
   val hbaseSiteXMLHDFSPathKey: String = "gimel.hbase.site.xml.hdfs.path"
+  val hbaseMaxRecordsForSchema: String = "gimel.hbase.schema.max.records"
+  val hbaseMaxColumnsForSchema: String = "gimel.hbase.schema.max.columns"
```
Is this max columns across all the namespaces or per namespace? Can we make this more explicit?

anishanainani replied:
Its for every dataset. As one dataset is based on a particular namespace and table, it will be based on a particular hbase table.

Dee-Pac commented on gimel-dataapi/gimel-connectors/gimel-hbase-1.2/src/main/scala/com/paypal/gimel/hbase/conf/HbaseConstants.scala:
```diff
@@ -29,4 +29,7 @@ object HbaseConstants {
   val putOperation = "put"
```
Could we retrofix and make all the variables UPPER_CASE?

anishanainani replied:
Sure

Dee-Pac commented on gimel-dataapi/gimel-connectors/gimel-hbase-1.2/src/main/scala/com/paypal/gimel/hbase/utilities/HBaseScanner.scala:
```diff
@@ -41,21 +42,27 @@ class HBaseScanner() {
   /**
     * Returns schema of hbase table
     *
+    * @param connection HBase Connection
```
NIT - in the method documentations, we do not need a ":" for each parameter.

Dee-Pac commented on gimel-dataapi/gimel-connectors/gimel-hbase-1.2/src/test/scala/com/paypal/gimel/hbase/DataSetTest.scala:
```diff
@@ -53,45 +53,45 @@ class DataSetTest extends HBaseLocalClient with Matchers {
-    }
-    assert(exception.getMessage.contains("HBase column family to columns mapping not found.") == true)
-  }
+//  test("Read operation - missing " + HbaseConfigs.hbaseColumnMappingKey) {
```
can we deleted the commented blocks of code?

anishanainani replied:
Sure

Dee-Pac commented:
@anishanainani I've completed my review. Most of the code looks clean. Please address the few comments that I've posted, and then we should be good to go.

Dee-Pac commented on gimel-dataapi/gimel-connectors/gimel-hbase-1.2/src/main/scala/com/paypal/gimel/hbase/utilities/HBaseScanner.scala:
```diff
@@ -196,8 +196,8 @@ class HBaseScanner() {
     *
-    * @param namespace : Name of hbase name space
-    * @param tableName : Name of hbase table
+    * @param namespace
```
@anishanainani  Did we delete the comments accidentally?

anishanainani replied:
I removed some of the param descriptions for which it was not necessary like tableName is self explanatory.

Dee-Pac commented:
@anishanainani  Thanks for addressing the review comments.

- Please address the last question.
- Also, could you rebase your commits?

anishanainani commented:
> @anishanainani Thanks for addressing the review comments.
> 
> * Please address the last question.
> * Also, could you rebase your commits?

If you do squash and commit, it automatically becomes 1 commit. Thats why I dint squash it manually.
